packages/core/src/parsers/jsx.ts

```diff
@@ -163,6 +163,25 @@ const parseCodeJson = (node: babel.types.Node) => {
   return tryParseJson(code);
 };
 
+const getPropsTypeRef = (node: babel.types.FunctionDeclaration): string => {
+  const param = (node.params[0] || {}) as babel.types.Identifier;
+  let propsTypeRef = '';
+  // TODO: component function params name must be props
+  if (
+    param.name === 'props' &&
+    babel.types.isTSTypeAnnotation(param.typeAnnotation)
+  ) {
+    const paramIdentifier = babel.types.variableDeclaration('let', [
+      babel.types.variableDeclarator(param),
+    ]);
+    propsTypeRef = generate(paramIdentifier)
+      .code.replace(/^let\sprops:\s+/, '')
+      .replace(/;/g, '');
+  }
+
+  return propsTypeRef;
+};
+
```

Comment:
Would be best to:
- avoid the type-cast, and check that `node.params[0]` is actually an Identifier
- return `undefined` when there is no type, instead of an empty string

```suggestion
const getPropsTypeRef = (
  node: babel.types.FunctionDeclaration,
): string | undefined => {
  const param = node.params[0];
  // TODO: component function params name must be props
  if (
    babel.types.isIdentifier(param) &&
    param.name === 'props' &&
    babel.types.isTSTypeAnnotation(param.typeAnnotation)
  ) {
    const paramIdentifier = babel.types.variableDeclaration('let', [
      babel.types.variableDeclarator(param),
    ]);
    return generate(paramIdentifier)
      .code.replace(/^let\sprops:\s+/, '')
      .replace(/;/g, '');
  }

  return undefined;
};
```